diff --git a/cld-0.2.pc.in b/cld-0.2.pc.in
index 631896f..b39057d 100644
--- a/cld-0.2.pc.in
+++ b/cld-0.2.pc.in
@@ -5,7 +5,7 @@ includedir=@includedir@

 Name: libcld
 Description: Library controls and data for DAQ and control application development.
-Requires: glib-2.0 gee-1.0 libxml-2.0
+Requires: glib-2.0 gee-0.8 libxml-2.0
 Version: @VERSION@
 Libs: -L${libdir} -lcld-0.2 -lxml2
 Cflags: -I${includedir}/cld-0.2 -I${libdir}/cld-0.2/include -I/usr/include/libxml2
diff --git a/cld/Makefile.am b/cld/Makefile.am
index 21a8514..92df5b6 100644
--- a/cld/Makefile.am
+++ b/cld/Makefile.am
@@ -45,11 +45,11 @@ INCLUDES = \

 libcld_0_2_la_VALAFLAGS = \
    -H cld.h --vapi cld-0.2.vapi \
-   -h cld-internals.h -g -X -DDEBUG \
+   -h cld-internals.h -g -X -DDEBUG -D HAVE_GLIB232 \
    --internal-vapi cld-internals-0.2.vapi \
    --library cld-0.2 --gir Cld-0.2.gir \
    --vapidir=$(srcdir)/../vapi --pkg config \
-   --pkg glib-2.0 --pkg gee-1.0 --pkg libxml-2.0 --pkg json-glib-1.0 \
+   --pkg glib-2.0 --pkg gee-0.8 --pkg libxml-2.0 --pkg json-glib-1.0 \
    --pkg linux --pkg posix \
    $(VALAFLAGS) \
    $(NULL)
diff --git a/cld/builder.vala b/cld/builder.vala
index d32ef40..9131fd5 100644
--- a/cld/builder.vala
+++ b/cld/builder.vala
@@ -146,7 +146,7 @@ public class Cld.Builder : GLib.Object {
         Gee.List<Object> map_values = new Gee.ArrayList<Object> ();

         map_values.add_all (objects.values);
-        map_values.sort ((GLib.CompareFunc) Object.compare);
+        map_values.sort ((GLib.CompareDataFunc<Cld.Object>?) Object.compare);
         objects.clear ();
         foreach (Object object in map_values) {
             objects.set (object.id, object);
diff --git a/configure.ac b/configure.ac
index 7e3e210..465041c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -29,16 +29,16 @@ dnl ---------------------------------------------------------------------------
 dnl Mandatory Libraries
 dnl ---------------------------------------------------------------------------
 GLIB_REQUIRED=2.32.0
-PKG_CHECK_MODULES([GLIB], glib-2.0 >= $GLIB_REQUIRED gobject-2.0 >= $GLIB_REQUIRED gthread-2.0 >= $GLIB_REQUIRED,
-   [AC_DEFINE(HAVE_GLIB232, [], [GLib 2.32 available])],
+PKG_CHECK_MODULES([GLIB], glib-2.0 >= $GLIB_REQUIRED gobject-2.0 gthread-2.0,
+   [AC_DEFINE(HAVE_GLIB232, [1], [GLib 2.32 available])],
    [PKG_CHECK_MODULES([GLIB], [glib-2.0 gobject-2.0 gthread-2.0],,
        [AC_MSG_ERROR(glib-2.12 is required)])
    ])
 AC_SUBST(GLIB_CFLAGS)
 AC_SUBST(GLIB_LIBS)

-PKG_CHECK_MODULES([GEE], gee-1.0,,
-   AC_MSG_ERROR(gee-1.0 is required))
+PKG_CHECK_MODULES([GEE], gee-0.8,,
+   AC_MSG_ERROR(gee-0.8 is required))
 AC_SUBST(GEE_CFLAGS)
 AC_SUBST(GEE_LIBS)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index e7f4dc0..5aa597a 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -35,7 +35,7 @@ tests_DEPENDENCIES = \
 tests_VALAFLAGS = \
    --basedir $(top_srcdir) \
    --vapidir $(top_srcdir)/cld \
-   --pkg cld-0.2 --pkg gee-1.0 --pkg libxml-2.0 \
+   --pkg cld-0.2 --pkg gee-0.8 --pkg libxml-2.0 \
    $(NULL)

 tests_CPPFLAGS = \
